File: postman/sdks/Specs SDKs/Search Service/python/src/search_service_sdk/models/health_check_response.py
from pydantic import Field
from typing import Optional
from .utils.base_model import BaseModel


class HealthCheckResponse(BaseModel):
    """Response model for the `health-check` endpoint.

    :param remote_addr: The origin of the request.
    :type remote_addr: str
    :param commit_hash: The current docker image used by the service.
    :type commit_hash: str
    :param service_es_connected: Bool that indicates if the service is connected to elasticsearch or not.
    :type service_es_connected: bool
    """

    remote_addr: str = Field(description="The origin of the request.")
    commit_hash: str = Field(
        description="The current docker image used by the service."
    )
    service_es_connected: bool = Field(
        description="Bool that indicates if the service is connected to elasticsearch or not."
    )

File: postman/sdks/Specs SDKs/Search Service/typescript/src/services/service-endpoints/models/health-check-response.ts
import { z } from 'zod';

/**
 * Zod schema for the HealthCheckResponse model.
 * Defines the structure and validation rules for this data type.
 * This is the shape used in application code - what developers interact with.
 */
export const healthCheckResponse = z.lazy(() => {
  return z.object({
    remoteAddr: z.string(),
    commitHash: z.string(),
    serviceEsConnected: z.boolean(),
  });
});

/**
 * Response model for the `health-check` endpoint.
 * @typedef  {HealthCheckResponse} healthCheckResponse - Response model for the `health-check` endpoint. - Response model for the `health-check` endpoint.
 * @property {string} - The origin of the request.
 * @property {string} - The current docker image used by the service.
 * @property {boolean} - Bool that indicates if the service is connected to elasticsearch or not.
 */
export type HealthCheckResponse = z.infer<typeof healthCheckResponse>;

/**
 * Zod schema for mapping API responses to the HealthCheckResponse application shape.
 * Handles any property name transformations from the API schema.
 * If property names match the API schema exactly, this is identical to the application shape.
 */
export const healthCheckResponseResponse = z.lazy(() => {
  return z
    .object({
      remote_addr: z.string(),
      commit_hash: z.string(),
      service_es_connected: z.boolean(),
    })
    .transform((data) => ({
      remoteAddr: data['remote_addr'],
      commitHash: data['commit_hash'],
      serviceEsConnected: data['service_es_connected'],
    }));
});

/**
 * Zod schema for mapping the HealthCheckResponse application shape to API requests.
 * Handles any property name transformations required by the API schema.
 * If property names match the API schema exactly, this is identical to the application shape.
 */
export const healthCheckResponseRequest = z.lazy(() => {
  return z
    .object({
      remoteAddr: z.string(),
      commitHash: z.string(),
      serviceEsConnected: z.boolean(),
    })
    .transform((data) => ({
      remote_addr: data['remoteAddr'],
      commit_hash: data['commitHash'],
      service_es_connected: data['serviceEsConnected'],
    }));
});
